sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/ChatClientBuilder.java

```diff
@@ -143,37 +170,58 @@ public ChatClientBuilder configuration(Configuration configuration) {
     }
 
     /**
-     * Sets the {@link RetryPolicy} that is used when each request is sent.
+     * Create synchronous chat client applying CommunicationTokenCredential, UserAgentPolicy,
+     * RetryPolicy, and CookiePolicy.
+     * Additional HttpPolicies specified by additionalPolicies will be applied after them
      *
-     * @param retryPolicy User's retry policy applied to each request.
-     * @return The updated {@link ChatClientBuilder} object.
-     * @throws NullPointerException If the specified {@code retryPolicy} is null.
+     * @return ChatClient instance
      */
-    public ChatClientBuilder retryPolicy(RetryPolicy retryPolicy) {
-        this.retryPolicy = Objects.requireNonNull(retryPolicy, "The retry policy cannot be null");
-        return this;
+    public ChatClient buildChatClient() {
+        ChatAsyncClient asyncClient = buildAsyncChatClient();
+        return new ChatClient(asyncClient);
+    }
+
+    /**
+     * Create asynchronous chat client applying CommunicationTokenCredential, UserAgentPolicy,
+     * RetryPolicy, and CookiePolicy.
+     * Additional HttpPolicies specified by additionalPolicies will be applied after them
+     *
+     * @return ChatAsyncClient instance
+     */
+    public ChatAsyncClient buildAsyncChatClient() {
+        AzureCommunicationChatServiceImpl internalClient = createInternalClient();
+        return new ChatAsyncClient(internalClient);
     }
 
     /**
-     * Create synchronous client applying CommunicationTokenCredential, UserAgentPolicy,
+     * Create synchronous chat thread client applying CommunicationTokenCredential, UserAgentPolicy,
      * RetryPolicy, and CookiePolicy.
      * Additional HttpPolicies specified by additionalPolicies will be applied after them
      *
+     * @param chatThreadId The id of the thread.
+     *
      * @return ChatClient instance
      */
-    public ChatClient buildClient() {
-        ChatAsyncClient asyncClient = buildAsyncClient();
-        return new ChatClient(asyncClient);
+    public ChatThreadClient buildChatThreadClient(String chatThreadId) {
```

Comment:
It'll be better to have a different builder for ChatThreadClient. This class is ChatClientBuilder! Most of the settings for building ChatClient are irrelevant for this function.

Reply:
> Most of the settings for building ChatClient are irrelevant for this function.

Could you elaborate on this?